--- linux-3.14.1/arch/powerpc/kernel/vdso32/Makefile.orig	2014-04-14 13:50:10.000000000 +0000
+++ linux-3.14.1/arch/powerpc/kernel/vdso32/Makefile	2014-12-03 06:33:39.656256064 +0000
@@ -43,7 +43,8 @@
 
 # actual build commands
 quiet_cmd_vdso32ld = VDSO32L $@
-      cmd_vdso32ld = $(CROSS32CC) $(c_flags) -Wl,-T $^ -o $@
+      cmd_vdso32ld = $(CROSS32CC) $(c_flags) -Wl,-T $^ -o $@ \
+		     $(if $(AFTER_LINK),; $(AFTER_LINK))
 quiet_cmd_vdso32as = VDSO32A $@
       cmd_vdso32as = $(CROSS32CC) $(a_flags) -c -o $@ $<
 
--- linux-3.14.1/arch/powerpc/kernel/vdso64/Makefile.orig	2014-04-14 13:50:10.000000000 +0000
+++ linux-3.14.1/arch/powerpc/kernel/vdso64/Makefile	2014-12-03 06:33:39.660228155 +0000
@@ -36,7 +36,8 @@
 
 # actual build commands
 quiet_cmd_vdso64ld = VDSO64L $@
-      cmd_vdso64ld = $(CC) $(c_flags) -Wl,-T $^ -o $@
+      cmd_vdso64ld = $(CC) $(c_flags) -Wl,-T $^ -o $@ \
+		     $(if $(AFTER_LINK),; $(AFTER_LINK))
 quiet_cmd_vdso64as = VDSO64A $@
       cmd_vdso64as = $(CC) $(a_flags) -c -o $@ $<
 
--- linux-3.14.1/arch/s390/kernel/vdso32/Makefile.orig	2014-04-14 13:50:10.000000000 +0000
+++ linux-3.14.1/arch/s390/kernel/vdso32/Makefile	2014-12-03 06:33:39.660228155 +0000
@@ -43,7 +43,8 @@
 
 # actual build commands
 quiet_cmd_vdso32ld = VDSO32L $@
-      cmd_vdso32ld = $(CC) $(c_flags) -Wl,-T $^ -o $@
+      cmd_vdso32ld = $(CC) $(c_flags) -Wl,-T $^ -o $@ \
+		     $(if $(AFTER_LINK),; $(AFTER_LINK))
 quiet_cmd_vdso32as = VDSO32A $@
       cmd_vdso32as = $(CC) $(a_flags) -c -o $@ $<
 
--- linux-3.14.1/arch/s390/kernel/vdso64/Makefile.orig	2014-04-14 13:50:10.000000000 +0000
+++ linux-3.14.1/arch/s390/kernel/vdso64/Makefile	2014-12-03 06:33:39.660228155 +0000
@@ -43,7 +43,8 @@
 
 # actual build commands
 quiet_cmd_vdso64ld = VDSO64L $@
-      cmd_vdso64ld = $(CC) $(c_flags) -Wl,-T $^ -o $@
+      cmd_vdso64ld = $(CC) $(c_flags) -Wl,-T $^ -o $@ \
+		     $(if $(AFTER_LINK),; $(AFTER_LINK))
 quiet_cmd_vdso64as = VDSO64A $@
       cmd_vdso64as = $(CC) $(a_flags) -c -o $@ $<
 
--- linux-3.14.1/arch/x86/vdso/Makefile.orig	2014-04-14 13:50:10.000000000 +0000
+++ linux-3.14.1/arch/x86/vdso/Makefile	2014-12-03 06:33:39.660228155 +0000
@@ -178,8 +178,9 @@
 quiet_cmd_vdso = VDSO    $@
       cmd_vdso = $(CC) -nostdlib -o $@ \
 		       $(VDSO_LDFLAGS) $(VDSO_LDFLAGS_$(filter %.lds,$(^F))) \
-		       -Wl,-T,$(filter %.lds,$^) $(filter %.o,$^) && \
-		 sh $(srctree)/$(src)/checkundef.sh '$(NM)' '$@'
+		       -Wl,-T,$(filter %.lds,$^) $(filter %.o,$^) \
+		$(if $(AFTER_LINK),; $(AFTER_LINK)) && \
+		sh $(srctree)/$(src)/checkundef.sh '$(NM)' '$@'
 
 VDSO_LDFLAGS = -fPIC -shared $(call cc-ldoption, -Wl$(comma)--hash-style=sysv)
 GCOV_PROFILE := n
--- linux-3.14.1/scripts/link-vmlinux.sh.orig	2014-04-14 13:50:10.000000000 +0000
+++ linux-3.14.1/scripts/link-vmlinux.sh	2014-12-03 06:33:39.660228155 +0000
@@ -65,6 +65,10 @@
 			-lutil ${1}
 		rm -f linux
 	fi
+	if [ -n "${AFTER_LINK}" ]; then
+		/usr/lib/rpm/debugedit -b ${RPM_BUILD_DIR} -d /usr/src/debug -i ${2} \
+			> ${2}.id
+	fi
 }
 
 
